### DISCUSSION TITLE
Next/Image keeping the original image SRC for better SEO

### DISCUSSION THREAD

**Discussion by giacomozecchini** [vercel/next.js#60888] [Ideas]
### Goals

1. Improving the Search Friendliness of the Image Component, avoiding a complete recrawl of "new" images and preventing drops in ranking.
2. Keeping Image Optimization for modern browsers.

### Background

As elaborated in [this Merj blog article](https://merj.com/blog/optimising-nextjs-image-component-for-image-search), the current implementation of the Next.js Image component provides developers with automatic image optimization, including “Size Optimization”. The [current implementation](https://nextjs.org/docs/pages/api-reference/components/image) involves rewriting the SRC attribute of the IMG tag and adding the SRCSET attribute with optimised images of different sizes for various screen sizes and resolutions.

When using the Next.js Image component with [URL] as the original image path, the output looks similar to this:

```
<img
    alt="..."
    fetchpriority="..."
    loading="..."
    width="..."
    height="..."
    decoding="..."
    data-nimg=".."
    class="..."
    sizes="..."
    srcset="/_next/image?url=[URL]&amp;w=320&amp;q=75 320w,
        /_next/image?url=[URL]&amp;w=384&amp;q=75 384w,
        /_next/image?url=[URL]&amp;w=768&amp;q=75 768w,
        /_next/image?url=[URL]&amp;w=1200&amp;q=75 1200w"
    src="/_next/image?url=[URL]&amp;w=1200&amp;q=75"
>
```

Major search engines primarily use the IMG tag's SRC attribute as the primary reference for images. 

Legacy websites that migrate to Next.js may have images on content delivery networks that have been ranking well on search engines. The alteration of the image URLs prompts search engines to recrawl the "new" images, leading to the indexing process and often resulting in a drop in ranking. For businesses heavily reliant on image traffic, this can be a significant challenge.

### Proposal

To address the issue of instability, we propose allowing the developer to set a custom SRC attribute and use the image optimised URLs exclusively in the SRCSET attribute. This would have the following benefits:
1. Search engines continue to use their original images from the SRC attribute, avoiding a complete recrawl of "new" images and preventing drops in ranking.
2. Modern browsers utilise the new optimised SRCSET attribute URLs to download the best image for users.

By implementing this change, the desired output should be:

```
<img
    alt="..."
    fetchpriority="..."
    loading="..."
    width="..."
    height="..."
    decoding="..."
    data-nimg=".."
    class="..."
    sizes="..."
    srcset="/_next/image?url=[URL]&amp;w=320&amp;q=75 320w,
        /_next/image?url=[URL]&amp;w=384&amp;q=75 384w,
        /_next/image?url=[URL]&amp;w=768&amp;q=75 768w,
        /_next/image?url=[URL]&amp;w=1200&amp;q=75 1200w"
    src="[URL]"
>
```

**Compatibility and Impact**
While, in an ideal world, this change should be the default output of the Image Component, altering the default output without specifying an option could affect websites that already use the image component.

Therefore, it is suggested to introduce an option/parameter to override the existing SRC attribute functionality, enhancing the search friendliness of the component, and to document this option for developers.

**Comment by cramforce** [comment]
CC @styfle

**Comment by styfle** [comment]
> it is suggested to introduce an option/parameter to override the existing SRC attribute functionality

I like the opt-in solution 👍 

Do you have a suggestion for the new prop name? Maybe `seoSrc`?


### Input
```tsx
import img from '../images/selfies/IMG123.jpg'

<Image 
  src={img}
  seoSrc="/images/me"
/>
```

### Output

```html
<img
  width="256"
  height="256"
  loading="lazy"
  decoding="async"
  srcset="/_next/image?url=%2F_next%2Fstatic%2Fmedia%2Fme.97e00142.jpg&amp;w=256&amp;q=75 1x, /_next/image?url=%2F_next%2Fstatic%2Fmedia%2Fme.97e00142.jpg&amp;w=640&amp;q=75 2x"
  src="/images/me">
```

**Reply by giacomozecchini** [reply]
Yes, the input-output behavior is correct. This would work for local and remote images.

I'd suggest using a more generic prop name, such as **overrideSrc="..."**, as people may need it for other reasons than SEO. But I'm not particularly attached to any particular prop name and happy for you to choose the name that would make the more sense.

**Reply by styfle** [reply]
I created a PR, thanks!

- https://github.com/vercel/next.js/pull/64221

**Reply by giacomozecchini** [reply]
Thank you for solving this!

**Comment by Malikmal** [comment]
Any update on this ?
